python/pyarrow/ipc.pxi

```diff
@@ -158,9 +160,14 @@ cdef class IpcWriteOptions(_Weakrefable):
     def compression(self, value):
         if value is None:
             self.c_options.codec.reset()
-        else:
+        elif isinstance(value, str):
             self.c_options.codec = shared_ptr[CCodec](GetResultValue(
                 CCodec.Create(_ensure_compression(value))).release())
+        elif isinstance(value, Codec):
+            self.c_options.codec = (<Codec>value).wrapped
```

Comment:
Just a nit, but we could use `.unwrap()` here.

Reply:
There already was an `unwrap()` but it returned `CCodec*` and not `shared_ptr<CCoded>`.  I think I'd need to investigate where the current `unwrap()` was used.